server/log_operation_manager.go

```diff
@@ -325,29 +235,79 @@ func (l *LogOperationManager) masterFor(ctx context.Context, allIDs []int64) ([]
 		l.tracker = election.NewMasterTracker(allIDs)
 	}
 
-	// Synchronize the set of configured log IDs with those we are tracking mastership for.
+	// Synchronize the set of configured log IDs with those we are tracking
+	// mastership for.
 	for _, logID := range allIDs {
-		knownLogs.Set(1.0, strconv.FormatInt(logID, 10))
+		label := strconv.FormatInt(logID, 10)
+		knownLogs.Set(1.0, label)
 		if l.electionRunner[logID] != nil {
 			continue
 		}
 		glog.Infof("create master election goroutine for %v", logID)
-		innerCtx, cancel := context.WithCancel(ctx)
-		election, err := l.info.Registry.ElectionFactory.NewElection(innerCtx, logID)
+		el, err := l.info.Registry.ElectionFactory.NewElection(ctx, logID)
 		if err != nil {
-			cancel()
 			return nil, fmt.Errorf("failed to create election for %d: %v", logID, err)
 		}
-		l.electionRunner[logID] = &electionRunner{
-			logID:    logID,
-			info:     &l.info,
-			tracker:  l.tracker,
-			cancel:   cancel,
-			wg:       &l.runnerWG,
-			election: election,
-		}
+
+		runner := election.NewRunner(&l.info.ElectionConfig, el, l.info.TimeSource, label+": ")
+		l.electionRunner[logID] = runner
+
+		//   beMaster := func(ctx context.Context) error {
```

Comment:
Is this meant to be here? Duplicates one in runner.go. If intended as documentation it should maybe be moved outside to the function doc.